experimental/emulated_vault/opt/emulated-vault/vault-server

```diff
@@ -0,0 +1,249 @@
+            if cdnFind:
+                def cdnFilter(key,val):
+                    try:
+                        data = json.loads(val)
+                        return data['cdn']==cdnFind.group(1)
+                    except:
+                        return False
+                filters['cdn'] = cdnFilter
+
+            dsFind = re.search(".*q=deliveryservice:([^&]*).*", parsed_path.query)
+            if dsFind:
+                def dsFilter(key,val):
+                    try:
+                        data = json.loads(val)
+                        return data['deliveryservice']==dsFind.group(1)
+                    except:
+                        return False
+                filters['ds'] = dsFilter
+
+            keyFind = re.search(".*q=_yz_rk:([^&]*).*", parsed_path.query)
+            if keyFind:
+                def keyFilter(key):
+                    return fnmatch.fnmatch(os.path.basename(key), keyFind.group(1))
+                keyFilters['key-match'] = keyFilter
+
+            success, parameters = adaptor.searchParameters("riak/ssl/", keyFilters=keyFilters, filters=filters)
+            if not success:
+                self.send_response(503)
+                self.send_header('Content-Type', 'application/json')
+                self.end_headers()
+                self.wfile.write('{"Failure"}')
+                return
+
+            docs = []
+            for key,val in parameters.iteritems():
+                docs.append(json.loads(val))
+            toReturn = {"response":{"numFound":len(docs),"start":0, "docs":docs}}
+
+            self.send_response(200)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+            self.wfile.write(json.dumps(toReturn))
+            return
+
+        logger.info("GET %s", self.path)
+        success, value = adaptor.getParameter(parsed_path.path[1:])
+        if not success:
+            self.send_response(503)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+            self.wfile.write('{"Failure"}')
+            return
+
+        if value is None:
+            self.send_response(404)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+            self.wfile.write('{"Not found"}')
+            return
+
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+        self.wfile.write(value)
+
+    def _do_POST(self):
+        logger.info("POST %s", self.path)
+        parsed_path = urlparse.urlparse(self.path)
+        content_len = int(self.headers.getheader('content-length'))
+        post_body = self.rfile.read(content_len)
+        data = json.loads(post_body)
+        #mimic vault beahvior
+        if parsed_path.path.startswith('/riak/ssl/'):
+            for item in data['certificate']:
+                data['certificate.%s'%item]=data['certificate'][item]
```

Comment:
Seems like a good use-case for format strings, dict comprehension, and key/value dict iteration:
```python3
data = {f"certificate.{key}": value for key, value in data.get("certificate", {})} 
```